src/main/clojure/clara/rules/engine.cljc

```diff
@@ -329,6 +329,7 @@
 (defrecord AlphaNode [env children activation]
```

Comment:
I'd be inclined to add a test that the listener is called correctly since this would be a functional part of Clara's API.